api/src/main/java/jakarta/data/metamodel/Pattern.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Operator;
+
+
+/**
+ * Represents a pattern-based restriction for matching operations, encapsulating different
+ * options such as prefix, suffix, and substring matching. This implementation
+ * allows flexibility in creating pattern-based conditions directly as `Restriction` instances.
+ *
+ * <p>Example usage with metadata attributes:</p>
+ * <pre>
+ * // Case-sensitive exact match
+ * Restriction<Book> exactMatch = Pattern.like(_Book.title, "Jakarta Data");
+ *
+ * // Case-insensitive prefix match
+ * Restriction<Book> prefixIgnoreCase = Pattern.prefixedIgnoreCase(_Book.title, "Jak");
+ *
+ * // Case-sensitive suffix match
```

Comment:
Per prior comments, we do not guarantee case sensitive.
```suggestion
 * // Suffix match
```